File: DSA_Level-1/Basics Of Programming/String, String Builder and ArrayList/String With Difference Of Every Two Consecutive Characters/StringwithDifference_Solution.js

function stringWithDifference(str){
    let res="";
    for(let i=0;i<str.length-1;++i){
        let diff=str.charCodeAt(i+1)-str.charCodeAt(i);
        res+=str[i]+diff;
    }

    res+=str[str.length-1];

    return res;
}

console.log(stringWithDifference("pepCODinG"));

File: DSA_Level-1/Basics Of Programming/String, String Builder and ArrayList/String With Difference Of Every Two Consecutive Characters/StringwithDifference_Solution.java
import java.io.*;
import java.util.*;

public class Main {

	public static String solution(String str){
		StringBuilder res=new StringBuilder();
		
		res.append(str.charAt(0));
		
		for(int i=1;i<str.length();++i){
		    int diff=str.charAt(i)-str.charAt(i-1);
		    res.append(diff);
		    res.append(str.charAt(i));
		}
		
		return res.toString();
	}
	public static void main(String[] args) {
		Scanner scn = new Scanner(System.in);
		String str = scn.next();
		System.out.println(solution(str));
	}

}